builtin/logical/aws/path_config_root.go

```diff
@@ -139,46 +139,81 @@ func (b *backend) pathConfigRootRead(ctx context.Context, req *logical.Request,
 }
 
 func (b *backend) pathConfigRootWrite(ctx context.Context, req *logical.Request, data *framework.FieldData) (*logical.Response, error) {
-	region := data.Get("region").(string)
-	iamendpoint := data.Get("iam_endpoint").(string)
-	stsendpoint := data.Get("sts_endpoint").(string)
-	stsregion := data.Get("sts_region").(string)
-	maxretries := data.Get("max_retries").(int)
-	roleARN := data.Get("role_arn").(string)
-	usernameTemplate := data.Get("username_template").(string)
-	if usernameTemplate == "" {
-		usernameTemplate = defaultUserNameTemplate
-	}
-
-	stsFallbackEndpoints := data.Get("sts_fallback_endpoints").([]string)
-	stsFallbackRegions := data.Get("sts_fallback_regions").([]string)
-
-	if len(stsFallbackEndpoints) != len(stsFallbackRegions) {
-		return logical.ErrorResponse("fallback endpoints and fallback regions must be the same length"), nil
-	}
-
 	b.clientMutex.Lock()
 	defer b.clientMutex.Unlock()
 
 	// check for existing config
-	previousCfg, previousCfgExists, err := getConfigFromStorage(ctx, req)
+	rc, err := getConfigFromStorage(ctx, req)
 	if err != nil {
 		return nil, err
 	}
 
-	rc := rootConfig{
-		AccessKey:            data.Get("access_key").(string),
-		SecretKey:            data.Get("secret_key").(string),
-		IAMEndpoint:          iamendpoint,
-		STSEndpoint:          stsendpoint,
-		STSRegion:            stsregion,
-		STSFallbackEndpoints: stsFallbackEndpoints,
-		STSFallbackRegions:   stsFallbackRegions,
-		Region:               region,
-		MaxRetries:           maxretries,
-		UsernameTemplate:     usernameTemplate,
-		RoleARN:              roleARN,
+	if rc == nil {
+		// Baseline
+		rc = &rootConfig{
+			AccessKey: data.Get("access_key").(string),
+			SecretKey: data.Get("secret_key").(string),
+		}
+	}
+
+	if region, ok := data.GetOk("region"); ok {
```

Comment:
Does updating the data extraction like this change any of the user-facing behaviors?